project/src/main/java/caselab/service/DocumentTypesService.java

```diff
@@ -1,26 +1,52 @@
 package caselab.service;
 
+import caselab.controller.types.payload.DocumentTypeDTO;
 import caselab.domain.entity.DocumentType;
 import caselab.domain.repository.DocumentTypesRepository;
 import java.util.Optional;
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatusCode;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
 
+@SuppressWarnings({"MagicNumber", "LineLength"})
 @Service
 @RequiredArgsConstructor
 public class DocumentTypesService {
     private final DocumentTypesRepository documentTypesRepository;
 
-    public Optional<DocumentType> findDocumentTypeById(Long id) {
-        return documentTypesRepository.findById(id);
+    public DocumentTypeDTO findDocumentTypeById(Long id) {
+        Optional<DocumentType> optionalDocumentType = documentTypesRepository.findById(id);
+        if (optionalDocumentType.isPresent()) {
+            return convertDocumentTypeToDocumentTypeDTO(optionalDocumentType.get());
+        } else {
+            throw HttpClientErrorException.NotFound.create(HttpStatusCode.valueOf(404), String.format("Тип документа с id= %s не найден", id), HttpHeaders.EMPTY, null, null);
+        }
     }
 
-    public DocumentType createDocumentType(DocumentType documentType) {
-        return documentTypesRepository.save(documentType);
+    public DocumentTypeDTO createDocumentType(DocumentTypeDTO documentTypeDTOForCreating) {
+        DocumentType documentTypeForCreating = convertDocumentTypeDTOToDocumentType(documentTypeDTOForCreating);
+        return convertDocumentTypeToDocumentTypeDTO(documentTypesRepository.save(documentTypeForCreating));
     }
 
     public void deleteDocumentTypeById(Long id) {
-        documentTypesRepository.deleteById(id);
+        Optional<DocumentType> optionalDocumentType = documentTypesRepository.findById(id);
+        if (optionalDocumentType.isPresent()) {
+            documentTypesRepository.deleteById(id);
+        }
+        throw HttpClientErrorException.NotFound.create(HttpStatusCode.valueOf(404),
```

Comment:
```suggestion
        var optionalDocumentType = documentTypesRepository.findById(id).orElseThrow(() -> new NoSuchElementException("No such element with id %d".formatted(id)));
            documentTypesRepository.deleteById(id);
```
А вообще в JPA вроде есть метод existsById